FullStorySegmentMiddleware/Classes/DictionaryHelper.m

```diff
@@ -0,0 +1,43 @@
+//
+//  DictionaryHelper.m
+//  Analytics
+//
+//  Created on 6/25/20.
+//  Copyright © 2020 FullStory. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "DictionaryHelper.h"
+
+@implementation DictionaryHelper
```

Comment:
I guess I was expecting that you'd create something like PropertiesDict which is derived from NSMutableDictionary.

Then in the calling code you'd do something like this:
```
PropertiesDict *props = [[PropertiesDict alloc] initWithCapacity:[properties count]];
[props append: withKey:[key stringByAppendingString:suffix] andSimpleObject:dict[key]];
```
I thought that there were several appendToDictionary variants?

Reply:
It used to have several, after I got rid of the original recursion logic, it simplified them to only needing one. So I moved the two related helper methods into this class. 
I read that iOS that prefers composition over subclassing: https://developer.apple.com/documentation/foundation/nsdictionary although I wasn't 100% sure what they mean by that. WDYT?

Reply:
Using composition would also be fine. For composition PropertiesDict would have an NSMutableDictionary as a member rather than as a parent. In this particular case, you really could go either way.

I don't see much value in having a helper class which is just static functions. You're really just using the class as a namespace then. I think that it would be better to either just make them free functions in the original file or by having a proper class (like PropertiesDict).

Reply:
Yup agree! I think it's going to be good to make it a proper class in case we need any more functions around the properties in the future.
^ now that I said that WDYT if we make the entire property transformation (`getSuffixedProps`) in this PR it's own class like [FSProperties] and function [fsProps getSuffixedPropsFromDictionary:properties], I think in the future we may provide more options like max search depth, pick or drop certain keys in the properties, etc.

Reply:
I think that would be good. However, the getSuffixedPropsFromDictionary function is like a factory function or constructor. getSuffixedProps doesn't reference self anywhere (except to call other methods), but because it generates what I was calling PropertyDict above. Looking at this some more, perhaps what it generates should really be something like an FSSuffixedProperties?

So you could have a FSSuffixedProperties class with appendToDictionary and pluralizeAllArrayKeysInDictionary as regular members and a static function called something like createSuffixedPropsFromDictionary which generates as output either an FSSuffixedProperties instance or perhaps just an NSDictionary and it uses FSSuffixedProperties to build that dictionary. It depends on whether you need to do additional things where having more member functions in FSSuffixedProperties might be useful. If you make createSuffixedPropsFromDictionary return just an NSDictionary, then I would make it be just a regular function (that happens to use FSSuffixedProperties to help it do itw work). If you make it return FSSuffixedProperties then it would be ok to make createSuffixedPropsFromDictionary be a constructor or static member function.